File: modules/common/sql/pagination.go
package sql

type Pagination struct {
	Limit  Int64Nullable
	Offset Int64Nullable
	Sort   StringNullable
}

type PaginationResult struct {
	Data      interface{}
	Total     int64
	TotalPage Int64Nullable
	Page      Int64Nullable
	Limit     Int64Nullable
}

File: resources/admin/types/pagination.ts
export type PaginationResult<T> = {
	data: T[];
	total: number;
	total_page?: number;
	page?: number;
	limit?: number;
};

export type GeneralResponse<T> = {
	status: string;
	message: string;
	detail?: string;
	data?: T;
};
